client/app/scripts/ng-sir-trevor-blocks.js

```diff
@@ -438,12 +439,15 @@ define([
                         contenteditable: true,
                         placeholder: that.descriptionPlaceholder
                     }).html(data.caption));
-                    
+
                     this.$editor.append($('<div>', {
                         name: 'credit',
                         class: 'st-image-block',
                         contenteditable: true,
-                        placeholder: that.authorPlaceholder
+                        placeholder: that.authorPlaceholder,
+                        oninput:
+                        'if(this.innerText.length > 300) { alert("' + that.limitedText + '"); this.innerText = this.innerText.substring(0, 300); this.style="border:1px solid red"}' +
```

Comment:
Adding 'oninput' attribute for the image author. oninput works for paste as well as opposed to keyup or keydown. As for the location and the form of the code, I did not find a better way to work with this sir trevor element.